edx_arch_experiments/kafka_consumer/management/commands/consume_events.py

```diff
@@ -103,17 +111,26 @@ def create_consumer(self, group_id):
 
         return DeserializingConsumer(consumer_config)
 
-    def handle_message(self, msg):
+    def emit_signals_from_message(self, msg):
         """
-        Place holder methods for how to handle an incoming message from the event bus
+        Determine the correct signal and send the event from the message
         """
-        # TODO (EventBus):
-        # Rewrite this to construct and/or emit the signal eventually specified in the message.
-        logger.info(f"Received message with key {msg.key()} and value {TrackingEvent.to_dict(msg.value())}")
+        if msg.headers():
+            event_types = [value for key, value in msg.headers() if key == EVENT_TYPE_HEADER]
```

Comment:
This seems strange. Why can't this use something like `if EVENT_TYPE_HEADER in msg.headers():` and `msg.headers()[EVENT_TYPE_HEADER]`?

Reply:
Thanks.
[nit] I think it would be cleaner to convert `event_types` to the more expected  `event_type` first, and then use `event_type` below. You could start with checks for `len == 0` (which I'd make an error because we can't proceed, rather than a warning), and `len > 1` (which I'd make a warning, because we can proceed).

Reply:
I mean, rather than a loop over event_types below, when I would expect a single event_type, I would suggest first resolving to the single event_type with appropriate error or warning for unexpected situations. Then below, just work with the known event_type.

At least for me, a loop seems unexpected and difficult to explain below, except if something has gone wrong. Unless I am missing something.

Reply:
We're talking about multiple event types for a single message. That should just not happen. I'm happy to have you fail hard on that case. Unless I am missing something. When do you think this might happen?

Reply:
I'm better with failing hard then with issuing a warning. I agree that at least for now we should just not let that happen unless someone makes a really good case later on.